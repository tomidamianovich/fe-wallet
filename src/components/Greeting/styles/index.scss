@import "../../../containers/App/styles/colours.scss";
@import "../../../containers/App/styles/responsiveness.scss";

.fe-wallet__greeting {
  display: flex;
  justify-content: space-between;
  &__heading {
    &__main {
      text-align: left;
      margin-bottom: 0.5rem;
    }
    &__sub {
      font-weight: 500;
      margin-inline-start: 0.05rem;
      margin-top: 0rem;
      align-self: flex-start;
    }
  }
  &__balance-container {
    margin-top: 1.5rem;
  }
}