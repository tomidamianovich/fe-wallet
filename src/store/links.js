const navLinks = [
  {
    name: 'Panel',
    to: 'panel',
    icon: 'panel'
  },
  {
    name: 'Billetera',
    to: 'wallet',
    icon: 'wallet'
  },
  {
    name: 'Cotizaciones',
    to: 'market-rates',
    icon: 'market-rates'
  },
  {
    name: 'Cr√©ditos',
    to: 'credit',
    icon: 'credit'
  },
  {
    name: 'Exchange',
    to: 'exchange',
    icon: 'exchange'
  },
  {
    name: 'Launchpad',
    to: 'launchpad',
    icon: 'launchpad'
  }
];

export default navLinks;