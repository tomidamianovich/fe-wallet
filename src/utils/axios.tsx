import { RateType, CurrencyType } from '../utils/type';
const axios = require('axios').default;

type responseType = {
  data: RateType[] | CurrencyType[]
}
  
const requestHandler = async (url:string): Promise<any> => 
  new Promise((resolve, reject) => 
    axios.get(url)
    .then((response: responseType) => resolve(response?.data))
    .catch((err:any) => reject(err))
  );

export {
  requestHandler,
}