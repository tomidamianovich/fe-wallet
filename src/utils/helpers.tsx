const handleFormatValues = (value:number) => value.toFixed(2);

export {
  handleFormatValues
}