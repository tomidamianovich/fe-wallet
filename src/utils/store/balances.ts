import { BalanceType } from '../type';

const balances:BalanceType[] = [
  {
    "ticker": "ARS",
    "amount": 98000,
    "account_limit": 10000000 
  },
  {
    "ticker": "USD",
    "amount": 0,
    "account_limit": 30000 
  },
  {
    "ticker": "BTC",
    "amount": 0.00948,
    "account_limit": 1 
  },
  {
    "ticker": "ETH",
    "amount": 1.23456,
    "account_limit": 30 
  },
  {
    "ticker": "DAI",
    "amount": 1.23456,
    "account_limit": 30000 
  },
  {
    "ticker": "RPC",
    "amount": 1.23456,
    "account_limit": 32820000 
  },
  {
    "ticker": "USDC",
    "amount": 1.23456,
    "account_limit": 300000 
  }
]


export default balances;