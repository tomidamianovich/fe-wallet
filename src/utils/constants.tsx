import { transaction, transactionTitle, transactionStatus } from '../utils/type';

const ENDPOINTS = {
  CURRENCIES: 'http://localhost:3001/currencies',
  RATES: 'http://localhost:3001/rates',
  BALANCES: 'http://localhost:3001/balance'
}

const PANEL = {
  HEADING: "Mi billetera",
  SUBHEADING: "En tu billetera vas a poder almacenar todas las criptomonedas que compres en ripio.",
}

const TRANSACTIONS:{
  HEADING: string,
  HEADER: string[],
  TYPES: transaction,
  TYPES_TITLES: transactionTitle,
  STATUS: transactionStatus
} = {
  HEADING: "Mis movimientos",
  HEADER: [
    "Fecha",
    "Tipo de Operación",
    "Método",
    "Estado",
    "Comisión",
    "Monto",
  ],
  TYPES: {
    DEPOSIT: "DEPOSIT",
    WITHDRAWAL: "WITHDRAWAL",
    SEND: "SEND",
    RECEIVE: "RECEIVE",
    BUY: "BUY",
    SWAP: "SWAP"
  },
  TYPES_TITLES: {
    DEPOSIT: "Ingreso",
    WITHDRAWAL: "Retiro",
    SEND: "Envio",
    RECEIVE: "Recibo",
    BUY: "Compra",
    SWAP: "Swap"
  },
  STATUS: {
    PENDING: "PENDING",
    FAILURE: "FAILURE",
    SUCCESS: "SUCCESS"
  }
}

const DEFAULT_BALANCE = {
  ticker: "",
  amount: 0,
  account_limit: 0,
  name: "",
  symbol: "",
  decimals: 2,
  type: "",
  actions: [{
    action: "",
    enabled: true
  }],
  url_images: {
    image_png: "",
    image_svg: "",
  },
  color: "",
  categories: [],
  is_favorite: true,
  sell_rate: ""
}

const COMISIONS = {
  BUY: 0.003,
  DEPOSIT: 0,
  WITHDRAWAL: 0.0004,
  SEND: 0,
  RECEIVE: 0,
  SWAP: 0.00004
}

type Link = {
  name: string;
  to: string;
  icon: string;
};

const NAV_LINKS:Link[] = [
  {
    name: 'Panel',
    to: 'panel',
    icon: 'panel'
  },
  {
    name: 'Billetera',
    to: 'wallet',
    icon: 'wallet'
  },
  {
    name: 'Cotizaciones',
    to: 'market-rates',
    icon: 'market-rates'
  },
  {
    name: 'Créditos',
    to: 'credit',
    icon: 'credit'
  },
  {
    name: 'Exchange',
    to: 'exchange',
    icon: 'exchange'
  },
  {
    name: 'Launchpad',
    to: 'launchpad',
    icon: 'launchpad'
  }
];

const WORDINGS = {
  BALANCE: {
    DETAIL: {
      INPUT: {
        LABELS: {
          TRANSACTION: "Seleccione el tipo de operacion a realizar",
          AMOUNT: "Ingrese cantidad",
          ARIA_LABEL : "Monto de la transacción"
        },
        MESSAGES: {
          INSUFFICIENT_AMOUNT: "Dinero insuficiente",
          AMOUNT_TO_DEBIT: "Monto a debitar",
          AMOUNT_NEEDED: "Monto necesario",
          ACTUAL_BALANCE: "Monto actual"
        }
      },
      ACTIONS: {
        CANCEL: {
          LABEL: "Cancelar",
          ARIA_LABEL: "Cancelar transacción",
        },
        CONFIRM: {
          LABEL: "Confirmar",
          ARIA_LABEL: "Confirmar transacción",
        }
      },
      CARD: {
        ACCOUNT: "Cuenta"
      }
    },
    SUMMARY: {
      DROPDOWN: {
        ARIA_LABEL: "Seleccionar moneda para ver balance"
      }
    }
  },
  CONVERTER: {
    FROM: {
      DROPDOWN: {
        ARIA_LABEL: "Seleccionar crypto origen",
        NAME: "from"
      },
      INPUT: {
        ARIA_LABEL: "Cripto a convertir",
        NAME: "from_amount"
      }
    },
    TO: {
      DROPDOWN: {
        ARIA_LABEL: "Seleccionar crypto destino",
        NAME: "to"
      },
      INPUT: {
        ARIA_LABEL: "Cripto convertida",
        NAME: "to_amount"
      }
    },
    BUTTON: {
      ARIA_LABEL: "Convertir",
      LABEL: "Convertir",
    }
  },
  CURRENCIES: {
    TYPES: {
      FIAT: "FIAT",
      CRYPTO: "CRYPTO"
    }
  },
  MAINTENANCE: {
    TITLE: "Pagina en Mantenimiento",
    SUBTITLE: {
      BEGINNING: "Oops! Estamos trabajando en la pagina ",
      ENDING: ". Ingrese nuevamente a esta seccion más tarde."
    }
  },
  NAVIGATION: {
    NAV_ITEM: {
      NAME: "Salir"
    }
  },
  NOT_FOUND: "Pagina no encontrada.",
  ERROR_FOUND: "Oops. Ocurrio un error",
  PAGES: {
    WALLET: "wallet",
    CREDIT: "credit",
    EXCHANGE: "exchange",
    LAUNCHPAD: "launchpad",
    MARKET_RATES: "market-rates",
    PANEL: "panel",
  }
}

export {
  ENDPOINTS,
  PANEL,
  TRANSACTIONS,
  DEFAULT_BALANCE,
  COMISIONS,
  NAV_LINKS,
  WORDINGS
}