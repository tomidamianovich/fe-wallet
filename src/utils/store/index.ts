import currencies from './currencies';
import cryptoCurrencies from './cryptoCurrencies';
import fiatCurrencies from './fiatCurrencies';
import balances from './balances';
import rates from './rates';

export {
  currencies,
  fiatCurrencies,
  cryptoCurrencies,
  rates,
  balances
}