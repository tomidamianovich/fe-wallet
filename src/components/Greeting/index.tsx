import React from 'react';
import { BalanceType, CurrencyType } from '../../utils/type';
import './styles/index.scss';
import { PANEL } from '../../utils/constants';
import BalanceSummary from '../BalanceSummary'

type Props = {
  balances: BalanceType[],
  currencies: CurrencyType[]
};

/*

  Component Name: Greeting
  Usages: 
    - Component that allows an user to see a greeting message
    - Component that shows BalanceSummary
*/

const Greeting: React.FC<Props> = ({
    balances,
    currencies
  }) => {
    const { HEADING, SUBHEADING } = PANEL;
    return (
      <div className="fe-wallet__greeting">
        <div className="fe-wallet__greeting__balance-container">
          <BalanceSummary balances={balances} currencies={currencies} />
        </div>
        <div className="fe-wallet__greeting__heading">
          <h2 className="fe-wallet__greeting__heading__main">{ HEADING }</h2>
          <h5 className="fe-wallet__greeting__heading__sub">{ SUBHEADING }</h5>
        </div>
      </div>  
    )
  }
;

export default Greeting;