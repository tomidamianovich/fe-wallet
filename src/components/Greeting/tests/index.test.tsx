import { render, screen, fireEvent, waitFor } from '@testing-library/react';
import Greeting from '../index';
import { balances, currencies } from '../../../utils/store';

test('renders Greeting with balances', () => {
  const { asFragment } = render(
    <Greeting 
      balances={balances}
      currencies={currencies}
    />
  );
  expect(asFragment()).toMatchSnapshot();
});

test('renders Greeting without balances', () => {
  const { asFragment } = render(
    <Greeting 
      balances={[]}
      currencies={[]}
    />
  );
  expect(asFragment()).toMatchSnapshot();
});

