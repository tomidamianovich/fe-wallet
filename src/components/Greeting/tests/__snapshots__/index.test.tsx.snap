// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Greeting with balances 1`] = `
<DocumentFragment>
  <div
    class="fe-wallet__greeting"
  >
    <div
      class="fe-wallet__greeting__heading"
    >
      <h2
        class="fe-wallet__greeting__heading__main"
      >
        Mi billetera
      </h2>
      <h5
        class="fe-wallet__greeting__heading__sub"
      >
        En tu billetera vas a poder almacenar todas las criptomonedas que compres en ripio.
      </h5>
    </div>
    <div
      class="fe-wallet__greeting__balance-container"
    >
      <div
        class="fe-wallet__balance"
      >
        <span
          class="fe-wallet__balance__symbol"
        >
          $
        </span>
        <span
          class="fe-wallet__balance__amount-desc"
        >
          98000.00
        </span>
        <select
          aria-label="Seleccionar moneda para ver balance"
          class="fe-wallet__dropdown fe-wallet__dropdown__balance"
          id="balance"
          name="balance"
        >
          <option
            value="0"
          >
            ARS
          </option>
          <option
            value="1"
          >
            BTC
          </option>
          <option
            value="2"
          >
            ETH
          </option>
          <option
            value="3"
          >
            USDC
          </option>
          <option
            value="4"
          >
            LTC
          </option>
          <option
            value="5"
          >
            DAI
          </option>
          <option
            value="6"
          >
            AXS
          </option>
          <option
            value="7"
          >
            BAT
          </option>
          <option
            value="8"
          >
            SOL
          </option>
          <option
            value="9"
          >
            LINK
          </option>
          <option
            value="10"
          >
            DOT
          </option>
          <option
            value="11"
          >
            MANA
          </option>
          <option
            value="12"
          >
            RPC
          </option>
          <option
            value="13"
          >
            SLP
          </option>
          <option
            value="14"
          >
            ADA
          </option>
          <option
            value="15"
          >
            UNI
          </option>
          <option
            value="16"
          >
            CHZ
          </option>
          <option
            value="17"
          >
            XRP
          </option>
          <option
            value="18"
          >
            USDT
          </option>
          <option
            value="19"
          >
            DOGE
          </option>
          <option
            value="20"
          >
            AAVE
          </option>
          <option
            value="21"
          >
            UBI
          </option>
        </select>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`renders Greeting without balances 1`] = `
<DocumentFragment>
  <div
    class="fe-wallet__greeting"
  >
    <div
      class="fe-wallet__greeting__heading"
    >
      <h2
        class="fe-wallet__greeting__heading__main"
      >
        Mi billetera
      </h2>
      <h5
        class="fe-wallet__greeting__heading__sub"
      >
        En tu billetera vas a poder almacenar todas las criptomonedas que compres en ripio.
      </h5>
    </div>
    <div
      class="fe-wallet__greeting__balance-container"
    >
      <div
        class="fe-wallet__balance"
      >
        <span
          class="fe-wallet__balance__symbol"
        />
        <span
          class="fe-wallet__balance__amount-desc"
        >
          NaN
        </span>
        <select
          aria-label="Seleccionar moneda para ver balance"
          class="fe-wallet__dropdown fe-wallet__dropdown__balance"
          id="balance"
          name="balance"
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;
